Source/Dafny/Compilers/Compiler-python.cs

```diff
@@ -0,0 +1,598 @@
+    }
+
+    protected override string GetCollectionBuilder_Build(CollectionType ct, IToken tok, string collName,
+      ConcreteSyntaxTree wr) {
+      throw new NotImplementedException();
+    }
+
+    protected override void EmitSingleValueGenerator(Expression e, bool inLetExprBody, string type,
+      ConcreteSyntaxTree wr) {
+      throw new NotImplementedException();
+    }
+
+    public override bool TextualTargetIsExecutable => true;
+
+    public override bool CompileTargetProgram(string dafnyProgramName, string targetProgramText,
+      string /*?*/ callToMain, string /*?*/ targetFilename, ReadOnlyCollection<string> otherFileNames,
+      bool runAfterCompile, TextWriter outputWriter, out object compilationResult) {
+      compilationResult = null;
+      if (runAfterCompile) {
+        Contract.Assert(callToMain != null); // this is part of the contract of CompileTargetProgram
+        // Since the program is to be run soon, nothing further is done here. Any compilation errors (that is, any errors
+        // in the emitted program--this should never happen if the compiler itself is correct) will be reported as 'node'
+        // will run the program.
+        return true;
+      } else {
+        // compile now
+        return SendToNewNodeProcess(dafnyProgramName, targetProgramText, null, targetFilename, otherFileNames,
+          outputWriter);
+      }
+    }
+
+    public override bool RunTargetProgram(string dafnyProgramName, string targetProgramText, string /*?*/ callToMain,
+      string targetFilename, ReadOnlyCollection<string> otherFileNames,
+      object compilationResult, TextWriter outputWriter) {
+
+      return SendToNewNodeProcess(dafnyProgramName, targetProgramText, callToMain, targetFilename, otherFileNames,
+        outputWriter);
+    }
+
+    bool SendToNewNodeProcess(string dafnyProgramName, string targetProgramText, string /*?*/ callToMain,
+      string targetFilename, ReadOnlyCollection<string> otherFileNames,
+      TextWriter outputWriter) {
+      Contract.Requires(targetFilename != null || otherFileNames.Count == 0);
+
+      var psi = new ProcessStartInfo("python", "") {
+        CreateNoWindow = true,
+        UseShellExecute = false,
+        RedirectStandardInput = true,
+        RedirectStandardOutput = false,
+        RedirectStandardError = false,
+      };
+
+      try {
+        using var nodeProcess = Process.Start(psi);
+        foreach (var filename in otherFileNames) {
+          WriteFromFile(filename, nodeProcess.StandardInput);
+        }
+
+        nodeProcess.StandardInput.Write(targetProgramText);
+        if (callToMain != null && DafnyOptions.O.RunAfterCompile) {
+          nodeProcess.StandardInput.Write(callToMain);
+        }
+
+        nodeProcess.StandardInput.Flush();
+        nodeProcess.StandardInput.Close();
+        nodeProcess.WaitForExit();
+        return nodeProcess.ExitCode == 0;
+      } catch (System.ComponentModel.Win32Exception e) {
```

Comment:
What would be exceptions that would run on Linux or MacOS ? Probably not Win32Exception. I can't help you with that but you can ask around.

Reply:
In his defence this is exactly the same as what the Javascript compiler catches for some reason. :)

But I think it would be much better to just catch `Exception` instead.